src/components/PopupWithForm.js

```diff
@@ -16,20 +26,24 @@ class PopupWithForm extends Popup {
     return formValues;
   }
 
+  open() {
+    super.open();
+    this.renderLoading(false);
```

Comment:
You need to return the **default** button text only in blocks `finally` so that the button could get the text back in any case: `success` or `failure`.